src/credentials/CHIPOperationalCredentials.cpp

```diff
@@ -266,6 +270,82 @@ CHIP_ERROR OperationalCredentialSet::SetDevOpCredKeypair(const CertificateKeyId
 }
 
+CHIP_ERROR OperationalCredentialSet::ToSerializable(const CertificateKeyId & trustedRootId,
+                                                    OperationalCredentialSerializable & serializable)
+{
+    const NodeCredential * nodeCredential = GetNodeCredentialAt(trustedRootId);
+    P256Keypair * keypair                 = GetNodeKeypairAt(trustedRootId);
+    P256SerializedKeypair serializedKeypair;
+    ChipCertificateSet * certificateSet  = FindCertSet(trustedRootId);
+    const ChipCertificateData * dataSet  = nullptr;
+    uint8_t * ptrSerializableCerts[]     = { serializable.mRootCertificate, serializable.mCACertificate,
+                                         serializable.mLeafCertificate };
+    uint16_t * ptrSerializableCertsLen[] = { &serializable.mRootCertificateLen, &serializable.mCACertificateLen,
+                                             &serializable.mLeafCertificateLen };
+
+    VerifyOrReturnError(certificateSet != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
+    ReturnErrorOnFailure(keypair->Serialize(serializedKeypair));
+    VerifyOrReturnError(serializedKeypair.Length() <= sizeof(serializable.mNodeKeypair), CHIP_ERROR_INVALID_ARGUMENT);
+
+    dataSet = certificateSet->GetCertSet();
+    VerifyOrReturnError(dataSet != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
+
+    memset(&serializable, 0, sizeof(serializable));
+    serializable.mNodeCredentialLen = nodeCredential->mLen;
+
+    memcpy(serializable.mNodeCredential, nodeCredential->mCredential, nodeCredential->mLen);
+    memcpy(serializable.mNodeKeypair, serializedKeypair, serializedKeypair.Length());
+    serializable.mNodeKeypairLen = static_cast<uint16_t>(serializedKeypair.Length());
+
+    for (uint8_t i = 0; i < certificateSet->GetCertCount(); ++i)
+    {
+        VerifyOrReturnError(CanCastTo<uint16_t>(dataSet[i].mCertificate.size()), CHIP_ERROR_INTERNAL);
+        memcpy(ptrSerializableCerts[i], dataSet[i].mCertificate.data(), dataSet[i].mCertificate.size());
+        *ptrSerializableCertsLen[i] = static_cast<uint16_t>(dataSet[i].mCertificate.size());
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR OperationalCredentialSet::FromSerializable(const OperationalCredentialSerializable & serializable)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    P256Keypair keypair;
+    P256SerializedKeypair serializedKeypair;
+    ChipCertificateSet certificateSet;
+    CertificateKeyId trustedRootId;
+
+    SuccessOrExit(err = certificateSet.Init(kOperationalCertificatesMax, kOperationalCertificateDecodeBufSize));
+
+    err = certificateSet.LoadCert(serializable.mRootCertificate, serializable.mRootCertificateLen,
+                                  BitFlags<CertDecodeFlags>(CertDecodeFlags::kIsTrustAnchor));
+    SuccessOrExit(err);
+
+    trustedRootId.mId  = certificateSet.GetLastCert()->mAuthKeyId.mId;
+    trustedRootId.mLen = certificateSet.GetLastCert()->mAuthKeyId.mLen;
+
+    err = certificateSet.LoadCert(serializable.mCACertificate, serializable.mCACertificateLen,
```

Comment:
What if CA certificate is not present?

Reply:
Updated this check to be optional as CA certificate is indeed not required to be present.